pyam/core.py

```diff
@@ -1302,6 +1336,14 @@ def _apply_criteria(df, criteria, **kwargs):
     return df
 
 
+def _make_index_from_data(df):
```

Comment:
this could probably be a bit more generic, like:

```
def _make_index(df, cols=['model', 'scenario']):
    return pd.MultiIndex.from_tuples(
       pd.unique(list(zip(df[col] for col in cols)),
       names = tuple(cols)
)
```

Reply:
nice, thanks @gidden!